edi_oca/models/edi_exchange_record.py

```diff
@@ -187,6 +200,20 @@ def _compute_ack_expected(self):
         for rec in self:
             rec.ack_expected = bool(self.type_id.ack_type_id)
 
+    @api.depends("related_record_ids")
+    def _compute_record(self):
+        """
+        Main related record is computed as the first related record created
+        for the exchange.
+        """
+        for rec in self:
```

Comment:
This could be expensive when you have tons of records.
IMO we should do a select for all records at once to get the 1st one right away.

Reply:
Have you tried w/ `limit=1` + `orderby="id"` ?
In any case we don't mind having sql here.